### ISSUE TITLE
i18n support how to

### ISSUE THREAD

**Issue by mckaygerhard** [hypery2k/owncloud#311]
how can i help with internationalization with russian, spanish and greek?

i only see the deustland languaje supoported and of course english

**Comment by mckaygerhard** [comment]
hey hey, calm u'r horses!!! i see a po file and a mo file, must i genretae the mo file with gettex , if so why then the php file?

**Comment by hypery2k** [comment]
you just need to add a new php file for each locale, copy the de.php for each locally ;)

**Comment by mckaygerhard** [comment]
added storqagechars2 es and ru texts with  064a80f9013c9419979b425bbbc9609d3523579f and started a wiki how to with two ways, the simple as added those files and another way such using translation web utility program interface or other..

**Comment by hypery2k** [comment]
thanks for the support

**Comment by hypery2k** [comment]
ignore the email from jenkins. I currently restructuring the tests!

**Comment by mckaygerhard** [comment]
made stup of wiki, work in progress https://github.com/hypery2k/owncloud/wiki/i18n-translations-internationalization-support-how-to

spanish and russian for roundcube and revealjs are in progress

**Comment by mckaygerhard** [comment]
due wiki are started i closed here, each i18n file will be a independent issue

**Comment by rotdrop** [comment]
Mmmh. Actually the getext template and intermediate translation .po-files are rather used to automatically add new text to the collection of strings or messages that should be translated. This gettext stuff is rather a method to keep thing in sync with the source-code, which is why I do not quite understand the point of removing the template files with commit  4e424d6f66db0a5361728bd4d4102fcf06cd2af3
Just a question. Just keeping the php-files and nothing else means that each new to-be-translated message has to be added manually to each translation file.

In this respect it should also be noted that in OC8 the php-files are no longer used: translations are converted into json format. There is a Perl-script in owncloud/l10n/l10n.pl (here "owncloud/" refers to the base-directory of the Owncloud installation, not to Martins OC-applications) which can be used to scan the source-code for messages and calls xgettext in order to update .pot and .po files.

One of the files removed by 4e424d6f66db0a5361728bd4d4102fcf06cd2af3 was a slightly tweaked version of that script, originating from the OC7 distribution.

Best,

Claus

**Comment by mckaygerhard** [comment]
umm that what's i hate when a enterprise leade and open source proyect! sure when i learn how to produce and pushed i18n files oc developer will change the api again!

**Comment by rotdrop** [comment]
Still this does not mean that gettext is useless; gettext is not an OC thing. Also, at least the i18n.pl script which converts the .po files to js and json is delivered together with the OC releases.

Regardless of the API issues, personally, I do not think that is a good idea to maintain hand-made translation files. I think that something like gettext pays off in the long run. Further, w.r.t. the API changes: if the po-files are used as the primary source for the translated messages, then API changes like this do not hurt so much.

Cheers,

Claus

**Comment by mckaygerhard** [comment]
i agree about mantain hand-made thinks, but its a system too complicated for few strings...

**Comment by mckaygerhard** [comment]
i changed the issue tittle, now here we have to discuss how we doit with i18n support and how we made that.. now post ideas and REMENBER easy to mantain and generated, we must take care about those apps must be easy to colaborate to others... and for author

**Comment by rotdrop** [comment]
OTOH, the rest of Owncloud uses the GNU gettext thing, which is also otherwise widely in use, at least in the open-source sector. Reinventing the wheel always bears the danger of finally ending up with something not less complicated and confusing as the original thing. Also, some authors work on more then just one App. So even if there are only few strings in Roundcube and StorageCharts, it might still be a good a idea to use a standard tool, or tools which are also used by other apps.

**Comment by mckaygerhard** [comment]
gettext as firts proposed.. ok...

note: i not part of mantain manually do it! but if are any more esyle rather that gettext i'll take it!

**Comment by hypery2k** [comment]
see https://github.com/hypery2k/owncloud/wiki/Localization for more details

**Comment by mckaygerhard** [comment]
je je very funny great script @hypery2k i can see that uses gettex so the system are then gettex, so the po file must be in sync with the sources, that part are by u'r own.. 

users part are only run the script, so the script i think (due i can see) generates or updates files using the po template?

**Comment by mckaygerhard** [comment]
hi @hypery2k i uploaded a firts translation file, this time for spanish ony, russian will waith for my (very bad) professor (@nerv55) to revise if i commit well the strings jeje

but i got and error wehen goes to write the php file... got this:

```
intranet@sysmail00:~/Devel/owncloud/roundcube/src/main/php/l10n$ ../../../../../scripts/l10n.pl roundcube write
Can't locate Locale/PO.pm in @INC (@INC contains: /etc/perl /usr/local/lib/perl/5.10.1 /usr/local/share/perl/5.10.1 /usr/lib/perl5 /usr/share/perl5 /usr/lib/perl/5.10 /usr/share/perl/5.10 /usr/local/lib/site_perl .) at ../../../../../scripts/l10n.pl line 3.
BEGIN failed--compilation aborted at ../../../../../scripts/l10n.pl line 3.
```

i soved installing in debian squeeze/lenny liblocale-po-perl package so i might document this in wiki, but  this work only for debian based, for any other distro each users must be by own

**Comment by mckaygerhard** [comment]
edited and added requeriments if got errors:

https://github.com/hypery2k/owncloud/wiki/Localization#requeriments

**Comment by mckaygerhard** [comment]
DOES storagechars2 app will stay as is or also a mechanism for language support will be generated? (i need what will be for close this issue)
